cluster-autoscaler/cloudprovider/azure/azure_util.go

```diff
@@ -647,3 +647,13 @@ func vmPowerStateFromStatuses(statuses []compute.InstanceViewStatus) string {
 	// PowerState is not set if the VM is still creating (or has failed creation)
 	return vmPowerStateUnknown
 }
+
+func getVMSSNameFromProviderID(providerID string) string {
+	parts := strings.Split(providerID, "/")
+	for i, part := range parts {
+		if part == "virtualMachineScaleSets" && i+1 < len(parts) {
+			return parts[i+1]
+		}
+	}
+	return ""
+}
```

Comment:
Ideally we could move to an implementation similar to the karpenter 


SubnetIDParts https://github.com/Azure/karpenter-provider-azure/blob/main/pkg/utils/subnet_parser.go#L38. 


We should also add to that structure some ID validation rather than having providerID validation sprinkled around the codebase. We should have one standard providerID validation bit. 


Not doing in this PR but just note it bothers me